tests/tensor_search/test_healthcheck.py

```diff
@@ -17,38 +18,776 @@ def tearDown(self) -> None:
         except IndexNotFoundError as e:
             pass
 
+    def create_mock_http_get(self, mock_health: dict = None, mock_settings: dict = None, mock_stats: dict = None):
+        """
+        Mocks calls to opensearch. Currently for health, we have 3: health, settings, and stats.
+        If a mock isn't provided, it will be set to the default from setUP.
+
+        Returns a function
+        """
+
+        def mock_http_get(path):
+            nonlocal mock_health, mock_settings, mock_stats
+            # mock settings should return GREEN, flood stage watermark NOT breached
+            # Inserting default mocks and also making deep copies so that the original mocks aren't modified each run
+            returned_mock_health = copy.deepcopy(mock_health) if mock_health is not None else {'status': 'green'}
+            returned_mock_settings = copy.deepcopy(mock_settings) if mock_settings is not None else {
+                "transient": {
+                    "cluster": {"routing": {"allocation": {"disk": {"watermark": {"flood_stage": "95%"}}}}}
+                },
+                "persistent": {},
+                "defaults": {}
```

Comment:
Can we have a test that has both all of defaults, transient and persistent, and make sure if honours the priority order? Perhaps also one where we have only defaults and persistent

Reply:
Yep, actually I have test cases for every combination of settings, as well as the watermark being in none/all of them. You can check them out here: https://github.com/marqo-ai/marqo/blob/joshua/opensearch-disk-breach/tests/tensor_search/test_healthcheck.py#L375